src/cryptoadvance/specter/util/version.py

```diff
@@ -12,12 +12,13 @@
 
 
 class VersionChecker:
-    def __init__(self, name="cryptoadvance.specter"):
+    def __init__(self, name="cryptoadvance.specter", specter=None):
```

Comment:
This is a good example: We're imho complicating here the specter-object even though this does not need to be necessarily tied to the specter-object. Maybe it make sense to have a "connection-object" which handles all sorts of connections. It can be passed to e.g. this VersionChecker and it would also potentially prevent cyclic dependencies which we already imho have in this case.